models/signed_latency.go

```diff
@@ -0,0 +1,148 @@
+package models
+
+import (
+	"crypto/ecdsa"
+	"github.com/aukilabs/go-tooling/pkg/errors"
+	"github.com/aukilabs/hagall-common/messages/hagallpb"
+	hwebsocket "github.com/aukilabs/hagall-common/websocket"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"sort"
+	"time"
+)
+
+type SignedLatency struct {
+	RequestID    uint32
+	StartedAt    time.Time
+	Iteration    uint32
+	PingRequests map[uint32]LatencyMetricsData
+	SessionID    string
+	ClientID     string
+	Address      string
+
+	sender     hwebsocket.ResponseSender
+	privateKey *ecdsa.PrivateKey
+
+	Min  float64
+	Max  float64
+	Mean float64
+	P95  float64
+}
+
+type LatencyMetricsData struct {
+	Start time.Time
+	End   time.Time
+}
+
+func (s *SignedLatency) Start(privateKey *ecdsa.PrivateKey, sender hwebsocket.ResponseSender, requestID, Iteration uint32, SessionId, ClientId, Address string) {
```

Comment:
The function parameters should start with lowercase. And ids should be written as `...ID`
Also, address should be walletAddress.